src/components/filterProducts/FilterProducts.js

```diff
@@ -0,0 +1,79 @@
+import { useLocation, Link } from "react-router-dom";
+import { useData } from "../../Context/stateManage/state-context";
+import ProductCard from "../productCards/ProductCard";
+import filterSearchProducts from "./filter-search-products";
+
+const FilterProducts = () => {
+  const {
+    state: { products },
+  } = useData();
+  const { search } = useLocation();
+
+  const searchParam = new URLSearchParams(search);
+
+  const searchQuery = searchParam.get("filter");
+
+  const filterProducts = filterSearchProducts(products, searchQuery);
+
+  return (
+    <main className='pd-container'>
+      {filterProducts.length > 0 ? (
+        <>
+          <div className='wishlist-head flex jc-center'>
+            <h4
+              className='pd-desc'
+              style={{
+                fontSize: "1.5rem",
+                textAlign: "center",
+              }}>
+              Found Items for "{searchQuery}" ({filterProducts.length})
+            </h4>
+          </div>
+          <section className='pd-container-main'>
+            <div className='pd-wrapper flex jc-center flex-gap flex-wrap'>
+              {filterProducts.map((product) => (
+                <ProductCard
+                  key={product._id}
+                  title={product.title}
+                  desc={product.desc}
+                  price={product.price}
+                  prevPrice={product.prev_price}
+                  discount={product.discount}
+                  img={product.img}
```

Comment:
Suggestion : pass product directly eg product={product} and then use the respective value in Product card